+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+classdef stewart < vdx.problems.Mpcc
+    properties
+        model
+        dcs
+        opts
+
+        populated
+    end
+
+    methods
+        function obj = stewart(dcs, opts)
+            obj = obj@vdx.problems.Mpcc();
+            obj.model = dcs.model;
+            obj.dcs = dcs;
+            obj.opts = opts;
+        end
+
+        function create_variables(obj)
+            dims = obj.dcs.dims;
+            dcs = obj.dcs;
+            model = obj.model;
+            opts = obj.opts;
+
+            obj.p.rho_h_p = {{'rho_h_p',1}, 1};
+            obj.p.rho_terminal_p = {{'rho_terminal_p',1}, 1};
+            obj.p.T = {{'T',1}, opts.T};
+            obj.p.p_global = {model.p_global, model.p_global_val};
+
+            % other derived values
+            h0 = opts.h_k;
+
+            % 0d vars
+            obj.w.v_global = {{'v_global',dims.n_v_global}, model.lbv_global, model.ubv_global, model.v0_global};
+            if opts.use_speed_of_time_variables && ~opts.local_speed_of_time_variable
+                obj.w.sot = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+            if opts.time_optimal_problem
+                obj.w.T_final = {{'T_final', 1}, opts.T_final_min, opts.T_final_max, opts.T}
+                obj.f = obj.f + obj.w.T_final();
+            end
+
+            % 1d vars
+            obj.w.u(1:opts.N_stages) = {{'u', dims.n_u}, model.lbu, model.ubu, model.u0};
+            obj.p.p_time_var(1:opts.N_stages) = {{'p_time_var', dims.n_p_time_var}, model.p_time_var_val};
+            if opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                obj.w.sot(1:opts.N_stages) = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+
+            % 2d vars
+            if obj.opts.use_fesd
+                ubh = (1 + opts.gamma_h) * h0;
+                lbh = (1 - opts.gamma_h) * h0;
+                if opts.time_rescaling && ~opts.use_speed_of_time_variables
+                    % if only time_rescaling is true, speed of time and step size all lumped together, e.g., \hat{h}_{k,i} = s_n * h_{k,i}, hence the bounds need to be extended.
+                    ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                    lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                end
+                obj.w.h(1:opts.N_stages,1:opts.N_finite_elements(1)) = {{'h', 1}, lbh, ubh, h0};
```

Comment:
opts.N_finite_elements(1) is a bit dirty. 
I know currently the different finite elements per stage (*) is broken - but maybe we make opts.N_finite_elements just a scalar again, and allow it to be a vector later when (*) is supported again - or we start supporting (*) right now

Reply:
I agree that it is ugly, however I have done a lot to improve performance of variable creation (to the point where it is 10x faster about for big problems to what we had before) in VDX which relies on this one-shot variable creation. I have a few ideas about how to maintain this speedup even with different numbers of finite elements but I believe this will take some backend work in VDX. 

I am updating to allow for variable `N_finite_elements` but adding a big todo comment to remind myself to do the backend work.